src/xpub.cpp

```diff
@@ -149,14 +149,20 @@ void zmq::xpub_t::xread_activated (pipe_t *pipe_)
             //  we can't give them back to userspace as it would be an API
             //  breakage since the payload of the message is completely
             //  different. Manually craft an old-style message instead.
-            data = data - 1;
-            size = size + 1;
+            //  Since with inproc we do not store the sub/cancel command
+            //  string anymore in the data buffer, there would be no room
+            //  to hack in the 0/1 byte, and it overwrites the metadata
+            //  pointer in the message struct, so create a new blob_t.
```

Comment:
Today, I still remember what you showed me yesterday, and what "hack in" and "overwrites the metadata pointer" means, but I guess this won't be understandable by anyone else or even by you or me in the not-too-distant future, so I guess it's better to remove this part entirely.

Reply:
That applies to the inproc case - I left it because for other transport, the data buffer does contain the command string so in theory it is possible to reuse it safely. I want to leave the comment there in case anybody in the future tries to do this optimization, as it's not immediately obvious. I'll clarify.

Reply:
Thanks for clarifying!